src/TSTransformer/util/createImportExpression.ts

```diff
@@ -87,25 +88,43 @@ function getNodeModulesImport(state: TransformState, moduleSpecifier: ts.Express
 
 	if (!moduleScope.startsWith("@")) {
 		DiagnosticService.addDiagnostic(errors.noUnscopedModule(moduleSpecifier));
-		return luau.none();
+		return [];
+	}
+
+	if (gameRbxPath && !gameRbxPath.includes(moduleScope)) {
+		DiagnosticService.addDiagnostic(
+			errors.noPackageImportWithoutScope(
+				moduleSpecifier,
+				path.relative(state.data.projectPath, moduleOutPath),
+				gameRbxPath,
+			),
+		);
+		return [];
 	}
 
 	const moduleName = relativeRbxPath[0];
 	assert(moduleName && typeof moduleName === "string");
 
 	if (!validateModule(state, moduleScope)) {
 		DiagnosticService.addDiagnostic(errors.noInvalidModule(moduleSpecifier));
-		return luau.none();
+		return [];
 	}
 
-	return propertyAccessExpressionChain(
-		luau.call(state.TS(moduleSpecifier.parent, "getModule"), [
-			luau.globals.script,
-			luau.string(moduleScope),
-			luau.string(moduleName),
-		]),
-		relativeRbxPath.slice(1),
-	);
+	if (state.projectType === ProjectType.Package) {
+		return [
+			propertyAccessExpressionChain(
+				luau.call(state.TS(moduleSpecifier.parent, "getModuleRelative"), [
+					luau.globals.script,
+					luau.string(moduleScope),
+					luau.string(moduleName),
+				]),
+				relativeRbxPath.slice(1),
+			),
+		];
+	} else {
+		assert(gameRbxPath);
```

Comment:
This assert is not necessary if the if statements above are broken into slightly different guards. That'll also help with readability.